app/models/work.rb

```diff
@@ -233,15 +233,17 @@ def expire_caches
     Work.expire_work_tag_groups_id(self.id)
   end
 
-  def update_pseud_index
-    return unless should_reindex_pseuds?
+  def update_pseud_and_collection_index
+    return unless should_reindex_pseuds_and_collections?
+
     IndexQueue.enqueue_ids(Pseud, pseud_ids, :background)
+    IndexQueue.enqueue_ids(Collection, collection_ids, :background)
```

Comment:
I think we also need to reindex any parent collection here as well, since the work count for parents includes the works for subcollections. 

(h/t ticking for [mentioning a similar issue on the CollectionItems model](https://github.com/otwcode/otwarchive/pull/3912#discussion_r528185995), which is now resolved.)